### ISSUE TITLE
BUG: `MemoryFS.move` fails to change filename.

### ISSUE THREAD

**Issue by tfeldmann** [PyFilesystem/pyfilesystem2#522]
I've got some strange behavior that I cannot explain.

I'm creating a file `Old-name.txt` on a MemoryFS, then moving it to a new location under a new name.
Problem is it still has the old name.

Testcase:

```python 
import fs
from fs.move import move_file


with fs.open_fs("mem://") as mem:
    old = mem.makedir("Old")
    new = mem.makedir("New")

    old.writetext("Old-name.txt", "Some content")

    print("Before:")
    mem.tree()

    move_file(
        src_fs=mem,
        src_path="Old/Old-name.txt",
        dst_fs=mem,
        dst_path="New/New-name.txt",
    )

    print("After: (Still the old name)")
    mem.tree()
```

Output:
```
Before:
├── New
└── Old
    └── Old-name.txt
After: (Still the old name)
├── New
│   └── Old-name.txt
└── Old
```

**Comment by tfeldmann** [comment]
Ok it's even more basic than that:

```python
import fs

with fs.open_fs("mem://") as mem:
    mem.writetext("Old-name.txt", "Some content")
    mem.move("Old-name.txt", "New-name.txt")
    mem.tree()
```

Output:
```
└── Old-name.txt
```

**Comment by althonos** [comment]
This is a known issue I fixed in #510 but I failed to do a new release for that sorry!

**Comment by tfeldmann** [comment]
Ah I see! Thank you for the fast response 👍

**Comment by althonos** [comment]
This has hopefully been fixed in `v2.4.15` :+1:

**Comment by tfeldmann** [comment]
Thanks a lot!
